File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> set_c;
        int max_len = 0;

        for (int i = 0; i < s.size(); ++i) {
            set_c.clear();
            int help = 0;
            for (int j = i; j < s.size(); ++j) {
                if (set_c.count(s[j])) break;
                set_c.insert(s[j]);
                help++;

            }
            max_len = max(max_len, help);
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        b, s = 0, 0
        mp = 0

        while s < len(prices):
            if prices[s] > prices[b]:
                profit =  prices[s] - prices[b]
                mp = max(mp, profit)
            else:
                b = s
            s += 1
        return mp
        